### PR TITLE
Add Cloud Foundry Userguide

### PR DESCRIPTION
**Pull request by orferch**
Provide a User Guide for the Cloud Foundry Plugin

### CODE REVIEW

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+- `CF_Instance_Api`: The endpoint of the provider
+- `CF_Instance_Organization`: The organization of the user account which should be used to deploy
+- `CF_Instance_Password`: Password of CloudFoundry user account
+- `CF_Instance_Space`: The space of the user account which should be used to deploy
+
+The credentials are used to get a list of services which can be used for deployment.
+For more details on supported NoteTypes, basics of Cloud Foundry, the Transformation and created scripts, see the documents in `dev/plugins/cloudFoundry`.
+
+## Artifact
+The Artifact is the result of the transformation which contains all necessary files which are needed to deploy an application on Cloud Foundry. After you open the Artifact you can see the log of the Transformation, the app folder(s) which contain your application and the output folder which contains the files for deployment. If you provide multiple applications in your CSAR the artifact will contain multiple app folders called app1, app2 and so on.
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: Not sure if enough context is provided for the term `application`. I know what you mean ofc-- maybe you could link to a section which explains what the plugin means exactly with that term,

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+![Artifact overview](img/artifact-overview.png)
+
+App folder: The .bp-config folder in this case contains buildpack additions for a php application, which needs some additional buildpacks.
+
+![Artifact app folder](img/artifact-app.png)
+
+The output folder contains a scripts folder and the files all_services.txt, manifest.yml, README.txt.
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: `, README.txt` -> `and README.txt`

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+![Artifact output folder](img/artifact-output.png)
+
+Scripts folder:
+
+![Artifact scripts folder](img/artifact-scripts.png)
+
+The Plugin tries to find free services available on the Cloud Foundry Platform to provide support for your modeled NodeTypes which are declared as services. They can be changed to paid plans. To see what additions a paid plan brings, please check your Cloud Foundry provider. You can see a list of all available services and corresponding plans in all_services.txt in the output folder.
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: `modeled NodeTypes`: I'd say `modeled NodeTemplates`.
`which are declared as services`: I would omit this.

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+For example if you want to change a database plan for the service cleardb (available on Pivotal Cloud Foundry), open `deploy.sh` in output/scripts/ and change `cf create-service cleardb spark my_db` from the free spark plan to boost, amp or shock: `cf create-service cleardb shock my_db`. For more information about the output folder see README.txt.
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: `For example if` -> `E.g., if`

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+## Requirements for Deployment
+Before you can use the artifact to deploy, there are some requirements:
+- Cloud Foundry is running on target infrastructure
+- the Cloud Foundry CLI is installed on the local machine
+Linux installation (other Systems: [Install CF CLI](https://docs.cloudfoundry.org/cf-cli/install-go-cli.html)):
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: What follows is an installation guide for debian-based linux systems, not for linux systems in general - it relies on `apt`.

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+```
+wget -q -O - https://packages.cloudfoundry.org/debian/cli.cloudfoundry.org.key | sudo apt-key add -
+echo "deb https://packages.cloudfoundry.org/debian stable main" | sudo tee /etc/apt/sources.list.d/cloudfoundry-cli.list
+sudo apt-get update
+sudo apt-get install cf-cli
+```
+
+- python 2.7 or newer is installed on the local machine
+- If the application contains databases, there needs to be an environment variable with the name `database_host` in the CSAR manifest.
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: In TOSCA slang, the 'manifest' is called `service template`.

Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: As a user, this requirement would leave me puzzled. Could you elaborate on that?

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+## Deployment
+To deploy an application, log into Cloud Foundry with the Cloud Foundry CLI (more information [Here](https://docs.cloudfoundry.org/cf-cli/getting-started.html)).
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: `Here` -> `here`

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+```
+cf login -a https://api.example.com -u username@example.com
+```
+
+- Unzip at target location
+
+- Move into the /output/scripts/ folder and execute `deploy.sh`:
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: `Move` describes the act of moving files. I guess you mean `Navigate`.

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+
+- lamp-input
+- lamp-multinode
+- lamp-noinput
+- task-translator
+
+These CSARs can be found in `TOSCAna/server/src/test/resources/csars/yaml/valid`
+
+## Doing a Transformation with a Test-CSAR
+The Lamp-input CSAR is used in this example, make sure you have done all the things from `Requirements for Deployment`. For details on how to use the GUI, see the Webapp User guide.
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: `example, make` -> `example. Make`

Comment (hnicke) [reply docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: Link to the webapp user guide is missing.

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+After you upload the CSAR to the GUI and select the Cloud Foundry platform as the target you get the following inputs screen (we use Pivotal CF as endpoint here):
+
+![CloudFoundry Inputs](img/inputs.png)
+
+Enter the needed inputs, click on Transform, wait for the Transformation to finish and download the artifact afterwards.
+Now follow the steps described under `Deployment` and shortly after the test application should be running.
```
Comment (hnicke) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: Did you know that one can use links to sections? `[Deployment](#deployment)` <- this should work.

**Comment by orferch** [comment]
The hardcoded Pivotal is a possible addition in the future, to support different providers with different services. I made a comment in the document that it was tested mainly on Pivotal.

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+The output folder contains a scripts folder and the files all_services.txt, manifest.yml and README.txt.
+
+![Artifact output folder](img/artifact-output.png)
+
+Scripts folder:
+
+![Artifact scripts folder](img/artifact-scripts.png)
+
+The Plugin tries to find free services available on the Cloud Foundry Platform to provide support for your modeled NodeTemplates. They can be changed to paid plans. To see what additions a paid plan brings, please check your Cloud Foundry provider. You can see a list of all available services and corresponding plans in all_services.txt in the output folder.
```
Comment (JensMueller2709) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: Consider using:
"To see what additions a paid plan brings..." -> "What benefits will bring changing to a paid plan..."

File: docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+
+E.g., if you want to change a database plan for the service cleardb (available on Pivotal Cloud Foundry), open `deploy.sh` in output/scripts/ and change `cf create-service cleardb spark my_db` from the free spark plan to boost, amp or shock: `cf create-service cleardb shock my_db`. For more information about the output folder see README.txt.
+
+## Requirements for Deployment
+Before you can use the artifact to deploy, there are some requirements:
+- Cloud Foundry is running on target infrastructure
+- the CLI output is in English:
```
Comment (JensMueller2709) [inline docs/user/plugins/cloudFoundry/cloud-foundry-plugin.md]: Maybe adding "if you are not sure use:"

**Comment by nfode** [comment]
@orferch please resolve the conflicts

**Pull request merged**